plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -46,8 +46,14 @@ local function filter(text, element)
         "div#newsstorytext",
         "div.general",
         }
-    if element and element ~= "" then
-        table.insert(selectors, 1, element)
+    if type(element) == "string" and element ~= "" then
+        table.insert(selectors, 1, element)  -- Insert string at the beginning
+    elseif type(element) == "table" then
+        for _, el in ipairs(element) do
+            if el ~= "" then
```

Comment:
I imagine that should also do the string check?

Reply:
Yes, but it's not doing that here.

Reply:
You're talking about line 49. I'm talking about line 53 not doing what line 49 does.

Reply:
I was indeed referring to invalid input, like:
```lua
filter_element = {{1}}
```
But it looks like it doesn't really cause any trouble further down the line. I'd still keep it `if type(el) == "string" and el ~= "" then` though for the sake of consistency.